[Jing Jie] iP

jjtan444 opened the pull request:
# Duke ChatBot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,
1. text-based
2. easy to learn
3. ~FAST~ _SUPER_ FAST to use

And it is **FREE**!

Features:
 - [x] Managing tasks
 - [ ]  Managing deadlines (coming soon)
 - [ ]  Reminders (coming soon)

### If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke("tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

jjtan444 committed:
Level-1

jjtan444 committed:
Level-2

jjtan444 committed:
Level-3

jjtan444 committed:
Level-4

jjtan444 committed:
A-TextUiTesting

jjtan444 committed:
Level-5

jjtan444 committed:
Level-6

jjtan444 committed:
Level-7

jjtan444 committed:
Revert "Level-7"

This reverts commit 626a0a907b667b68e22f5075b894cb2dd3e46f4a.

jjtan444 committed:
Revert "Revert "Level-7""

This reverts commit 824586ac9539bf4f5b33b36874782401a8402e28.

jjtan444 committed:
To level 7

jjtan444 committed:
Back to Level 6

This reverts commit e31fadcb37fa08515e2ac779a212c76bd2eed66f.

jjtan444 committed:
Merge branch 'branch-Level-7'

jjtan444 committed:
To level 8

jjtan444 committed:
Merge with branch Level 8

jjtan444 committed:
A-MoreOOP

jjtan444 committed:
A-Packages

jjtan444 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

jjtan444 committed:
A-Gradle

jjtan444 committed:
A-JUnit

jjtan444 committed:
A-JavaDoc

jjtan444 committed:
A-Coding-Standard

jjtan444 committed:
Merge branch 'branch-A-CodingStandard'

jjtan444 committed:
Level-9

jjtan444 committed:
Merge branch 'branch-Level-9'

gnahzeus commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+        return this.date.format(this.OUTPUT_FORMAT);
+    }
+
+    private static LocalDateTime dateTimeParser(String time) throws DukeException {
```
Should this method be named as a verb instead, e.g. `parseDateTime()`?

shogun187 replied:
Maybe name this method as a verb?

gnahzeus commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,99 @@
+        this.filePath = filepath;
+    }
+
+    /**
+     * Saves the tasklist.
+     * @param items the stringified tasklist.
+     */
```
Might it be better to have an empty line between the description and parameter section?

gnahzeus commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+
+    /**
+     * Converts list to list of string.
+     * @return
```
Should there be a description for the return value?

gnahzeus commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    /**
+     * Prints  List message.
```
Might be good to remove the extra space in this line!

gnahzeus reviewed (commented):
Hey @jjtan444, aside from a few nits with the naming, this LGTM! :+1:

jjtan444 committed:
Level-10

shogun187 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,160 @@
+    /**
+     * Creates a Duke instance.
+     * @param filePath the fileName.
+     */
```
Shouldn't there be an empty line between description and parameter section?

shogun187 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,160 @@
+     * Gets a response from Ui.
+     * @param input the command.
+     * @return the output.
+     */
```
Shouldn't there be an empty line between description and parameter section?

shogun187 commented on src/main/java/Duke/DukeException.java:
```diff
@@ -0,0 +1,11 @@
+    /**
+     * Creates Duke exception instance.
+     * @param message error to print.
+     */
```
Shouldn't there be an empty line between description and parameter section?

shogun187 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,53 @@
+
+public class Parser {
+
+    /**
```
Shouldn't there be an empty line between description and parameter section?

shogun187 commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,99 @@
+
+    private String filePath;
+
+    /**
```
Shouldn't there be an empty line between description and parameter section?

shogun187 commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    /**
+     * Loads the file.
```
Shouldn't there be an empty line between description and parameter section?

shogun187 commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,64 @@
+    private String descript;
+
+    /**
+     * Creates the Task instance.
```
Shouldn't there be an empty line between description and parameter section?

shogun187 reviewed (commented):
Looks good. Keep it up man! 👍 
Just minor nitpicks.

jjtan444 committed:
Added checkstyle

jjtan444 committed:
Added assertions

jjtan444 committed:
Improved code quality

jjtan444 committed:
Merge pull request #1 from jjtan444/branch-A-Assertions

Added assertions

jjtan444 committed:
Merge branch 'master' into branch-A-CodeQuality

jjtan444 committed:
Merge pull request #2 from jjtan444/branch-A-CodeQuality

Improved code quality

jjtan444 committed:
Ui class does not check for duplicates while adding tasks.

Being able to check for duplicates prevents conflicts.

This commit ensures there are no duplicates before adding tasks.

jjtan444 committed:
When user gives "bye" input nothing happens

It is not a preferred action as it does not work as expected
as the app does not close as the user intended.

The app now closes after a "bye" input.

jjtan444 committed:
Refactored Storage loadfile

jjtan444 committed:
Update README.md

jjtan444 committed:
Added User Guide in README.md

jjtan444 committed:
Added Ui.png

jjtan444 committed:
Added Ui.png image

jjtan444 committed:
Delete Ui.PNG

jjtan444 committed:
Delete Ui.png.PNG

jjtan444 committed:
Added Ui.png

jjtan444 committed:
Refactored code and improved code quality

jjtan444 committed:
Merge branch 'master' of https://github.com/jjtan444/iP

jjtan444 committed:
Update README.md

jjtan444 committed:
Update README.md

jjtan444 committed:
Further improved code quality

jjtan444 committed:
Merge branch 'master' of https://github.com/jjtan444/iP
